### PR TITLE
Add BCL functionality

### PR DESCRIPTION
**Pull request by macumber**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,8 @@
 /test/
 /lib/measures/test_results
 /lib/measures/.rubocop*
+/lib/measures/staged
```
[the remark this answers is not in the document]
Comment (kflemin) [reply .gitignore]: I can move them wherever, just let me know!

File: lib/openstudio/extension/rake_task.rb [hunk 2]
```diff
@@ -133,14 +134,238 @@ def setup_subtasks(name)
...
+            if ENV['bcl_download_uid']
+              uid = ENV['bcl_download_uid']
+            else
+              uid = '7c65af63-4316-47c9-b7a5-68d7a019e533'
+            end
+
+            content = bcl.download_component(uid)
+
+            # save as tar.gz
+            download_path = @files_dir + '/downloads'
```
[the remark this answers is not in the document]
Comment (kflemin) [reply lib/openstudio/extension/rake_task.rb]: sure, i'll move it.

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (nllong) [reply .gitignore]: Hey @kflemin -- i know this was *ages* ago. Can we create a new folder at the root called "staging" or something similar?

**Comment by nllong** [comment]
Can we review this and merge if all tests out?

**Comment by kflemin** [comment]
yes @nllong this was ages ago!  I think we can review and move the folders to wherever.  If I remember correctly, this was a first-pass attempt

**Comment by nllong** [comment]
@kflemin can we schedule someone to work on this? Would be nice to get working in openstudio 3.x

**Pull request merged**